lib/fog/aws/requests/compute/request_spot_instances.rb

```diff
@@ -71,7 +71,7 @@ def request_spot_instances(image_id, instance_type, spot_price, options = {})
             options.merge!(Fog::AWS.indexed_param('LaunchSpecification.SecurityGroupId', [*security_group_ids]))
           end
           if options['LaunchSpecification.UserData']
-            options['LaunchSpecification.UserData'] = Base64.encode64(options['LaunchSpecification.UserData'])
+            options['LaunchSpecification.UserData'] = Base64.encode64(options['LaunchSpecification.UserData']).chomp!
```

Comment:
Interesting.  I can't get 1.8 to install any more so I'm going to take your word for it.